api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -115,6 +115,10 @@ type HostedControlPlaneSpec struct {
 	// +optional
 	KubeConfig *KubeconfigSecretRef `json:"kubeconfig,omitempty"`
 
+	// kubeAPICustomKubeconfig is a group that encompasses an additional KAS server address and the customkubeconfig generated for the kube-apiserver.
```

Comment:
Can we capture all the following:
customKubeconfig specifies the configuration to generate a kubeconfig resolving through a kasDNSName. For this kubeconfig to successfully resolve, this needs to be used in conjunction with hostedCluster.apiServer.ServingCerts and hostedCluster.apiServer.ClientCA. DNS for this name resolution needs to be setup out of band to resolve to the kas publishing strategy